### PR TITLE
Fix errors on missing contexts

### PR DESCRIPTION
**Pull request by sagesmith-wf**
#### Changes:

- Updated unit tests
- Fixed errors that occurred when when context was None
- Merged with Chris's changes

Please review: @hefischer  @andrewperkins-wf

### CODE REVIEW

**Comment by jasonleinbach-wf** [comment]
+1

**Comment by jasonleinbach-wf** [comment]
Original review: https://github.com/Workiva/dqc_us_rules/pull/7

File: dqc_us_rules/util/facts.py [hunk 1]
```diff
@@ -175,15 +175,36 @@ def LegalEntityAxis_facts_by_member(facts):
...
     for fact in facts:
         legalDim = LEGALENTITYAXIS_DEFAULT
-        dims = [dim for dim in fact.context.segDimValues.values() if dim.isExplicit and dim.member is not None]
-        for dim in dims:
-            if dim.dimension.qname.localName == 'LegalEntityAxis':
-                legalDim = dim.member.qname.localName
-                break
-        results[legalDim].append(fact)
+        if fact_components_valid(fact):
+            dims = [dim for dim in fact.context.segDimValues.values() if dim.isExplicit and dim.member is not None]
```
Comment (chrislococo-wf) [inline dqc_us_rules/util/facts.py]: SG: Please try to limit lines to 80 characters, suggestion:
```
dims = [
    dim for dim in fact.context.segDimValues.values() 
    if dim.isExplicit and dim.member is not None
]
```

File: dqc_us_rules/util/facts.py [hunk 2]
```diff
@@ -175,15 +175,36 @@ def LegalEntityAxis_facts_by_member(facts):
...
+            results[legalDim].append(fact)
     return results
 
 
+def fact_components_valid(fact):
+    """
+    Return true if all of the components in a fact are none
+
+    :param fact: The fact to check if it is valid
+    :type fact: arelle.ModelInstanceObject.ModelFact
```
Comment (chrislococo-wf) [inline dqc_us_rules/util/facts.py]: SG: Incorrect type formatting:
```
:type fact: :class:`~arelle.ModelInstanceObject.ModelFact`
```

File: dqc_us_rules/util/facts.py [hunk 3]
```diff
@@ -175,15 +175,36 @@ def LegalEntityAxis_facts_by_member(facts):
...
+                if dim.dimension.qname.localName == 'LegalEntityAxis':
+                    legalDim = dim.member.qname.localName
+                    break
+            results[legalDim].append(fact)
     return results
 
 
+def fact_components_valid(fact):
+    """
+    Return true if all of the components in a fact are none
```
Comment (brianmyers-wf) [inline dqc_us_rules/util/facts.py]: Think I've pointed this out before, but this docstring and the ```:return:``` string are misleading as you are returning ```True``` only if the fact components are NOT none.

**Comment by brianmyers-wf** [comment]
pass complete, good unit tests, one comment about a comment

File: dqc_us_rules/util/facts.py [hunk 4]
```diff
@@ -175,15 +175,36 @@ def LegalEntityAxis_facts_by_member(facts):
...
     return results
 
 
+def fact_components_valid(fact):
+    """
+    Return true if all of the components in a fact are not none
+
+    :param fact: The fact to check if it is valid
+    :type fact: '~arelle.ModelInstanceObject.ModelFact'
+    :return: True if none of the components of the fact are None
```
Comment (brianmyers-wf) [inline dqc_us_rules/util/facts.py]: This is also "not None".

File: dqc_us_rules/util/facts.py (hunk 1 above)
Comment (chrislococo-wf) [inline dqc_us_rules/util/facts.py]: Addressed in https://github.com/DataQualityCommittee/dqc_us_rules/pull/62

**Comment by brianmyers-wf** [comment]
+1

**Comment by jasonleinbach-wf** [comment]
+1

**Comment by chrislococo-wf** [comment]
+10 QA performed in the Arelle app using a modified XBRL document.

**Comment by andrewperkins-wf** [comment]
All reviews and testing have been performed; code looks good to me.
+1 and merging.

**Pull request merged**